modules/cluster_estimation/cluster_estimation.py

```diff
@@ -159,24 +168,95 @@ def run(
         # Store new input data
         self.__current_bucket += self.__convert_detections_to_point(detections)
 
+        # Decide to run
+        if not self.__decide_to_run(run_override):
+            return False, None
+
+        # sort bucket by label in descending order
+        self.__current_bucket = self.__sort_by_labels(self.__current_bucket)
```

Comment:
This needs to be `self.__all_points`, as `current_bucket` is emptied and moved over to `all_points` after `decide_to_run()`.